packages/salesforcedx-vscode-core/src/conflict/timestampConflictDetector.ts

```diff
@@ -54,6 +63,7 @@ export class TimestampConflictDetector {
       lastModifiedInCache = cache.getPropertiesForFile(key)?.lastModifiedDate;
       if (
         !lastModifiedInCache ||
+        skipTimestampCheck ||
```

Comment:
This loop both checks whether the file has a conflict (by checking timestamps) and then creates the diff for the file if it does have a conflict.

In the case of a SourceConflictError, this same function is being re-used to create the diffs, and the skipTimestampCheck option is used to skip the step where it checks the timestamps.  This seemed to keep the refactoring here to a minimum, but let me know if separating out these two functions (timestamp checking and diff creating) would make more sense here.